ntoskrnl/mm/ARM3/virtual.c

```diff
@@ -5472,8 +5484,8 @@ NtFreeVirtualMemory(IN HANDLE ProcessHandle,
                 //
                 if ((EndingAddress >> PAGE_SHIFT) == Vad->EndingVpn)
                 {
-                    PMEMORY_AREA MemoryArea;
-
+                    //
+                    // Case C (freeing a part at the end)
```

Comment:
Nit: also above `This is case B or case C. First check for case C` -> `This is case B or case C.`